src/Runner.Listener/SelfUpdaterV2.cs

```diff
@@ -0,0 +1,623 @@
+﻿using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.IO.Compression;
+using System.Linq;
+using System.Net.Http;
+using System.Reflection;
+using System.Security.Cryptography;
+using System.Threading;
+using System.Threading.Tasks;
+using GitHub.DistributedTask.WebApi;
+using GitHub.Runner.Common;
+using GitHub.Runner.Common.Util;
+using GitHub.Runner.Sdk;
+using GitHub.Services.Common;
+using GitHub.Services.WebApi;
+
+namespace GitHub.Runner.Listener
+{
+    public class SelfUpdaterV2 : RunnerService, ISelfUpdater
```

Comment:
Consider adding a comment here about SeflUpdaterV2 and how it differs from SelfUpdater (removes dependency on package service but requires package data in the refresh message).

Reply:
added